diff --git a/Papyrus/Source/Scripts/diffs/rn_utility_transfer.psc b/Papyrus/Source/Scripts/rn_utility_transfer.psc
index b174d52..76307c9 100644
--- a/Papyrus/Source/Scripts/diffs/rn_utility_transfer.psc
+++ b/Papyrus/Source/Scripts/rn_utility_transfer.psc
@@ -1,6 +1,7 @@
 Scriptname RN_Utility_Transfer extends Quest 
 
 Import RN_Utility_Global
+import JZCL_TCCHelperFunctions
 
 RN_PatchAPI property API auto
  
@@ -41,34 +42,8 @@ Bool Property SilentTransfer auto hidden
 Function RunCustomTransfer()
 	
 	DBM_SortWait.SetValue(1)
-	Transfered = 0
+	Transfered = CustomTransfer(DBM_AutoSortDropOff, API.TokenRefList_NoShipment)
 	
-	Int Index = API.TokenRefList_NoShipment.length
-	Bool Transferable
-	While Index
-		Index -= 1
-		if (API.TokenRefList_NoShipment[Index])
-			Notify("The Curators Companion: Transfering Items...", MCM.ColourString)
-			Int Index2 = API.TokenRefList_NoShipment[Index].GetNumItems()
-			While Index2
-				Index2 -= 1
-				Form ItemRelic = API.TokenRefList_NoShipment[Index].GetNthForm(Index2)
-				if dbmMaster.HasForm(ItemRelic) && !dbmDisp.HasForm(ItemRelic)
-					if API.TokenRefList_NoShipment[Index] as Actor
-						Actor _Actor = API.TokenRefList_NoShipment[Index] as Actor
-						Transferable = !_Actor.IsEquipped(ItemRelic) && !Game.IsObjectFavorited(ItemRelic) && !RN_ExcludedItems_Generic.HasForm(ItemRelic) && !DBM_ProtectedItems.HasForm(ItemRelic)
-					else
-						Transferable = !RN_ExcludedItems_Generic.HasForm(ItemRelic) && !DBM_ProtectedItems.HasForm(ItemRelic)
-					endIF
-					
-					if Transferable
-						API.TokenRefList_NoShipment[Index].RemoveItem(ItemRelic, 1, true, DBM_AutoSortDropOff)
-						Transfered += 1
-					endIf
-				endIf
-			endWhile
-		endIf
-	endWhile
 	DBM_SortWait.SetValue(0)
 	DisplayFunc()
 endFunction
@@ -76,35 +51,8 @@ endFunction
 Function RunAllTransfer()
 	
 	DBM_SortWait.SetValue(1)
-	Transfered = 0
+	Transfered = AllTransfer(PlayerRef, DBM_AutoSortDropOff, API.TokenRefList)
 
-	Int Index = API.TokenRefList.length
-	Bool Transferable
-	While Index
-		Index -= 1
-		ObjectReference _Container = API.TokenRefList[Index] as ObjectReference	
-		if _Container
-			Notify("The Curators Companion: Transfering Items...", MCM.ColourString)
-			Int Index2 = _Container.GetNumItems()
-			While Index2
-				Index2 -= 1
-				Form ItemRelic = _Container.GetNthForm(Index2)
-				if dbmMaster.HasForm(ItemRelic) && !dbmDisp.HasForm(ItemRelic)
-					if _Container as Actor && _Container != PlayerRef
-						Actor _Actor = _Container as Actor
-						Transferable = !_Actor.IsEquipped(ItemRelic) && !Game.IsObjectFavorited(ItemRelic) && !RN_ExcludedItems_Generic.HasForm(ItemRelic) && !DBM_ProtectedItems.HasForm(ItemRelic)
-					else
-						Transferable = !Game.GetPlayer().IsEquipped(ItemRelic) && !Game.IsObjectFavorited(ItemRelic) && !RN_ExcludedItems_Generic.HasForm(ItemRelic) && !DBM_ProtectedItems.HasForm(ItemRelic)
-					endIF
-					
-					if Transferable
-						_Container.RemoveItem(ItemRelic, 1, true, DBM_AutoSortDropOff)
-						Transfered += 1
-					endIf
-				endIf
-			endWhile
-		endif
-	endWhile
 	DBM_SortWait.SetValue(0)
 	DisplayFunc()
 endFunction
@@ -112,21 +60,8 @@ endFunction
 Function RunRelicTransfer()
 	
 	DBM_SortWait.SetValue(1)
-	Transfered = 0
-	
-	Notify("The Curators Companion: Transfering Items...", MCM.ColourString)
-	Int Index = RN_Storage_Container.GetNumItems()
-	While Index
-		Index -= 1
-		Form ItemRelic = RN_Storage_Container.GetNthForm(Index)
-		if dbmMaster.HasForm(ItemRelic) && !dbmDisp.HasForm(ItemRelic)
-			Bool Transferable = !RN_ExcludedItems_Generic.HasForm(ItemRelic) && !DBM_ProtectedItems.HasForm(ItemRelic)
-			if Transferable
-				RN_Storage_Container.RemoveItem(ItemRelic, 1, true, DBM_AutoSortDropOff)
-				Transfered += 1
-			endIf
-		endIf
-	endWhile
+	Transfered = RelicTransfer(RN_Storage_Container, DBM_AutoSortDropOff)
+
 	DBM_SortWait.SetValue(0)
 	DisplayFunc()
 endFunction
